#430: Use proper plurals based on image settings

jamesmorrison opened the pull request:
### Description of the Change
Adds singular labels where a single image is selected for generation.

Closes #430

### How to test the Change

**Test 1: Plural**:
- Open Classifai settings:
  - set the number of images to generate between 2 - 10 (i.e. not 1)
  - Check / ensure: `Enable image generation` is checked
- Create a new post:
  - Open the media modal (in a new post => featured image / media block) and select `Generate Images`
  - Confirm the text remains as:
    - `Enter a prompt below to generate images.`
    - `Once images are generated, choose one or more of those to import into your Media Library and then choose one image to insert.`

**Test 3: Singular**:
- Open Classifai settings:
  - set the number of images to generate 1 (i.e. not 2 - 10)
  - Check / ensure: `Enable image generation` is checked
- Create a new post:
  - Open the media modal (in a new post => featured image / media block) and select `Generate Images`
  - Confirm the text shows singular terminology:
    - `Enter a prompt below to generate an image.` - new
    - `Once an image is generated, you can import it into your Media Library and then select to insert.` - new



### Changelog Entry
> Added - Add singular labels when a single image is selected for generation.

### Credits
Props @jamesmorrison


### Checklist:
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

jamesmorrison committed:
Use singluar / plural instructions based on number of images set to be generated automatically.

jamesmorrison commented:
@jeffpaul / @dkotter Please could you review this change? If you'd prefer alternative singular test, please let me know what this should be updated to - I added what I thought was appropriate here.

ravinderk commented on includes/Classifai/Providers/OpenAI/DallE.php:
```diff
@@ -128,16 +128,30 @@ public function enqueue_admin_scripts( $hook_suffix = '' ) {
 				<p>
-					<?php esc_html_e( 'Enter a prompt below to generate images.', 'classifai' ); ?>
+					<?php
+					if ( $number_of_images > 1 ) {
```
We can use [`_n`](https://developer.wordpress.org/reference/functions/_n/) function to handle plural translation.

jamesmorrison replied:
Thanks @ravinderk; I've updated this.

dkotter replied:
Apologies for not seeing this sooner but I don't think this is the approach we should take. While it is cleaner, some languages use the singular form for numbers other than 1 (see https://codex.wordpress.org/I18n_for_WordPress_Developers#Plurals). I suggest we go back to the original approach [here](https://github.com/10up/classifai/pull/482/commits/d097367e5c51bfc3d16ff86a3eeed76538c632cf)

jamesmorrison replied:
@dkotter Ok, I've reverted this.

ravinderk commented on includes/Classifai/Providers/OpenAI/DallE.php:
```diff
@@ -128,16 +128,30 @@ public function enqueue_admin_scripts( $hook_suffix = '' ) {
 				<p>
-					<?php esc_html_e( 'Once images are generated, choose one or more of those to import into your Media Library and then choose one image to insert.', 'classifai' ); ?>
+					<?php
+					if ( $number_of_images > 1 ) {
```
We use [`_n`](https://developer.wordpress.org/reference/functions/_n/) function to handle plural translation.

jamesmorrison replied:
Thanks @ravinderk; I've updated this.

jamesmorrison committed:
Switch to _n() function.

dkotter commented:
@jamesmorrison Thanks for the work here! I've left one comment that I think we should address. There's also a few other text strings that we probably want to update.

- The `Generate images` tab text: https://github.com/10up/classifai/blob/develop/includes/Classifai/Providers/OpenAI/DallE.php#L126
- The `Generate images` button text: https://github.com/10up/classifai/blob/develop/includes/Classifai/Providers/OpenAI/DallE.php#L150
- Text shown after an image is generated: https://github.com/10up/classifai/blob/develop/includes/Classifai/Providers/OpenAI/DallE.php#L154

jamesmorrison committed:
Reverted to if else check; added singular / plural text for missing string.

jamesmorrison commented:
@dkotter I've reverted to if else (rather than `_n()`) and added singular text where this was missed previously. I've tested and confirmed this works; please could you check this?

dkotter committed:
Update one more text string. Fix spacing

dkotter committed:
Merge branch 'develop' into feature/430-use-proper-plurals-based-on-image-setting
